parse.py

```diff
@@ -0,0 +1,209 @@
+from functions import common
+from functions import exponents_and_logs
+from functions import statistic
+from functions import trigonometry
+import exceptions.exceptions as exceptions
+
+from pyparsing import *
```

Comment:
maybe not a good idea as it will put irrelevant things into the code, also can cause name conflict

Reply:
The alternative is the long list of imports that we had before. I could change it back if you guys are really worried about the sanity of this approach but we haven't had any conflicts yet and the Python code seems pretty complete.

Reply:
I would say revert this to be thorough